barretenberg/cpp/src/barretenberg/stdlib/primitives/group/cycle_scalar.cpp

```diff
@@ -68,17 +65,14 @@ cycle_scalar<Builder> cycle_scalar<Builder>::from_witness(Builder* context, cons
  * @tparam Builder
  * @param context
  * @param value
- * @param num_bits
  * @return cycle_scalar<Builder>
  */
 template <typename Builder>
-cycle_scalar<Builder> cycle_scalar<Builder>::from_witness_bitstring(Builder* context,
-                                                                    const uint256_t& bitstring,
-                                                                    const size_t num_bits)
+cycle_scalar<Builder> cycle_scalar<Builder>::from_256_bit_witness(Builder* context, const uint256_t& bitstring)
 {
-    BB_ASSERT_LT(bitstring.get_msb(), num_bits);
+    const size_t num_bits = 256;
     const uint256_t lo_v = bitstring.slice(0, LO_BITS);
-    const uint256_t hi_v = bitstring.slice(LO_BITS, HI_BITS);
```

Comment:
If `slice()` had not been accidentally returning the high 128 bits, we would have detected the misuse of the interface here. In this case, we really do want the high 128 bits (instead of just the high 126) since the input is an arbitrary `uint256_t`. Yet we're passing `bitstring.slice(128, 126);`